### ISSUE TITLE
Change reducer parameters order

### ISSUE THREAD

**Issue by guitcastro** [evant/redux#7]
Redux framework defines reducers as:

`(previousState, action) => newState`

And all implementations I found for redux (Java, Kotlin, Swift, JS ...) the first argument is the state and the second one is the action, although this library use action as the first parameter and the state as the second. 

So I guess to make sense use the same parameters others from the oficial redux docs.

**Comment by evant** [comment]
Yeah, makes sense. Will have to figure out the backwards compatibility story around this if any.
